app/objects/c_agent.py

```diff
@@ -275,8 +278,34 @@ async def task(self, abilities, obfuscator, facts=(), deadman=False):
     def all_facts(self):
         return [f for lnk in self.links for f in lnk.facts if f.score > 0]
 
+    @property
+    def executor_change_to_assign(self):
+        return self._executor_change_to_assign
+
+    @executor_change_to_assign.setter
+    def executor_change_to_assign(self, executor_change_dict):
+        """Set pending executor change dict for the agent."""
+        if executor_change_dict:
+            self._executor_change_to_assign = executor_change_dict
+            if executor_change_dict.get('action') == 'remove':
```

Comment:
So, how is the dict going to be created?  Is it going to be something from a yaml ability file (user created), or something that we generate based on what is in a yaml file (auto generated)?  If user created we are opening it up to exceptions as people put in junk data or something not even a dictionary.  We also could handle this in marshmallow too on the loading of ability files, so that would work as well.

Reply:
Good question. I was originally planning on having custom planners be the primary source of setting executor updates. We could certainly expand it in the future to a GUI option where users can manually adjust agent executor information. I wasn't going to have this tied to ability files. I can enforce dict types too when setting it from a planner

Reply:
Sounds good, and true, if the initial use is going to be for removing an executor that is easily done at the planner level and structures managed there.

It is when adding an executor that the ability changes would possibly be needed for specifying a payload, or moving/changing names of executor binaries (encapsulating the platform specific commands to do so).